DivanteLtd/vue-storefront
# add lazy create cart token

### Short Description and Why It's Useful

This allows to create cart token when user add product to cart. There is edge case: when user load page, go to offline, add product, go to checkout => can't make order. That's because he hasn't got payment methods which are loaded after token creation. Also I've added refresh payment methods in checkout. Right now in demo if somehow user get to checkout without loaded payment-methods then he can't make order, even if he get back online

### QA
A.
1. clear cache
2. add product to cart
3. go to checkout
4. make order
5. Expected: user made order

B.
1. clear cache
2. go offline
3. add product to cart
4. check cart - there should be product
5. go to checkout
6. you should not make order
7. turn online
8. now you can make order

Same can be checked with user logged in.


### Which Environment This Relates To
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)


### Upgrade Notes and Changelog
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance
